linalg/ode.hpp

```diff
@@ -530,27 +407,164 @@ class ESDIRK33Solver : public ODESolver
 class GeneralizedAlphaSolver : public ODESolver
 {
 protected:
-   mutable Vector xdot,k,y;
+   mutable Vector k,y;
    double alpha_f, alpha_m, gamma;
-   int  nstate;
 
    void SetRhoInf(double rho_inf);
    void PrintProperties(std::ostream &out = mfem::out);
 public:
 
    GeneralizedAlphaSolver(double rho = 1.0) { SetRhoInf(rho); };
+   void Init(TimeDependentOperator &f_) override;
+   void Step(Vector &x, double &t, double &dt) override;
+
+};
+
+/** An explicit Adams-Bashforth method. */
+class AdamsBashforthSolver : public ODESolver
+{
+private:
+   const double *a;
+   const int stages;
+   double dt_;
+
+protected:
+   std::unique_ptr<ODESolver> RKsolver;
+
+   inline bool print()
+   {
+#ifdef MFEM_USE_MPI
+      return Mpi::IsInitialized() ? Mpi::Root() : true;
+#else
+      return true;
+#endif
+   }
+
+   void CheckTimestep(double dt);
 
+public:
+   AdamsBashforthSolver(int s_, const double *a_);
    void Init(TimeDependentOperator &f_) override;
+   void Step(Vector &x, double &t, double &dt) override;
+};
 
+/** A 1-stage, 1st order AB method.  */
+class AB1Solver : public AdamsBashforthSolver
+{
+private:
+   static MFEM_EXPORT const double a[1];
+
+public:
+   AB1Solver() : AdamsBashforthSolver(1, a) { RKsolver = nullptr; }
```

Comment:
The trivial constructor for `std::unique_ptr` will construct the pointer such that it evaluates to `nullptr` until it is assigned or reset... so this line should be able to be simplified to
```suggestion
   AB1Solver() : AdamsBashforthSolver(1, a) {}
```

Reply:
Done! Thanks.